maven-javadoc-plugin/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -4739,6 +4741,19 @@ private void addJavadocOptions( List<String> arguments, List<String> sourcePaths
         }
     }
 
+    private static File findMainDescriptor( List<String> roots )
+    {
+        for ( String root : roots )
+        {
+            File descriptorFile = new File( root, "module-info.java" ).getAbsoluteFile();
+            if ( descriptorFile.exists() )

```

Comment:
Won't be better isFile() instead of exists() ?

Reply:
I agree, I'll update shortly.

Reply:
Actually, I'm kind of inclined to fail hard and fast if something exists called `module-info.java` that isn't a regular file. That almost certainly indicates some sort of problem in the build elsewhere.

Reply:
Totally makes sense +1